File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */

 public class Solution 
 {
     public ListNode getIntersectionNode(ListNode headA, ListNode headB) 
     {
         var A = headA;
         var B = headB;
 
         while (A != null && B != null)
         {
             if (A == B)
                 return A;
             else
             {
                 if (A.next == null)
                 {
                     A = headA;
                     B = B.next;
                 }
                 else
                     A = A.next;
             }
         }    
 
         return null;
     }
 }

File: leetcode/100-999/160. Intersection of Two Linked Lists/solution_1.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */

var getIntersectionNode = function(headA, headB) 
{
    let A = headA
    let B = headB

    while (A && B)
    {
        if (A == B)
        {
            return A
        }
        else
        {
            if (A.next == null)
            {
                A = headA
                B = B.next
            }
            else
                A = A.next     
        }
    }

    return null
};